### ISSUE TITLE
milClassifier not compatible with OpenCV3

### ISSUE THREAD

**Issue by Tobias-Fischer** [robotology/iol#21]
Hi,
We recently updated some of our machines to OpenCV3. Thanks to @pattacini most repositories are compatible with this version. However, `milClassifier` does not compile with the new version:
```
[ 92%] Building CXX object src/milClassifier/CMakeFiles/milClassifier.dir/main.cpp.o
/home/icub/robot/iol/src/milClassifier/main.cpp: In constructor ‘MILPort::MILPort(yarp::os::ResourceFinder&)’:
/home/icub/robot/iol/src/milClassifier/main.cpp:818:17: error: ‘create’ is not a member of ‘cv::FeatureDetector {aka cv::Feature2D}’
        detector=cv::FeatureDetector::create(detector_type);
                 ^
/home/icub/robot/iol/src/milClassifier/main.cpp:819:18: error: ‘create’ is not a member of ‘cv::DescriptorExtractor {aka cv::Feature2D}’
        extractor=cv::DescriptorExtractor::create(extractor_type);
```

The corresponding lines in the source are these ones:
```
       string detector_type=bFeature.check("detector",Value(feature_name.c_str())).asString().c_str();
       string extractor_type=bFeature.check("extractor",Value(feature_name.c_str())).asString().c_str();

       detector=cv::FeatureDetector::create(detector_type);
       extractor=cv::DescriptorExtractor::create(extractor_type);
```

Unfortunately, creating instances by string is not possible in OpenCV3 anymore (see http://docs.opencv.org/3.1.0/db/dfa/tutorial_transition_guide.html). I am not sure what the best way will be to get around this.
Did you need to solve similar issues in the other repositories @pattacini?

Thanks,
Tobi

PS: This is not urgent, as far as I know `milClassifier` is not needed in the recent pipelines anymore.

**Comment by pattacini** [comment]
Thanks @Tobias-Fischer for the catch.
I'll be fixing those bugs in the next days.

**Comment by pattacini** [comment]
Easy fix via a4feafd5621cb3621237a325c917bfa1300f4ae3 :smirk: 
`milClassifier` is an old module, almost deprecated.

I've realized that also for other superseded module like `sparseCoder` we (~ I :smile:) decided to go for such policy.
